nucliadb/src/nucliadb/search/search/find_merge.py

```diff
@@ -325,47 +263,58 @@ def rank_fusion_merge(
+                order=0,  # NOTE: this will be filled later
+                text="",  # NOTE: this will be filled later too
+                position=TextPosition(
+                    page_number=paragraph.metadata.position.page_number,
+                    index=paragraph.metadata.position.index,
+                    start=paragraph.start,
+                    end=paragraph.end,
+                    start_seconds=[x for x in paragraph.metadata.position.start_seconds],
+                    end_seconds=[x for x in paragraph.metadata.position.end_seconds],
+                ),
+                paragraph_labels=[x for x in paragraph.labels],  # XXX could be list(paragraph.labels)?
+                fuzzy_search=fuzzy_result,
                 is_a_table=paragraph.metadata.representation.is_a_table,
+                representation_file=paragraph.metadata.representation.file,
+                page_with_visual=paragraph.metadata.page_with_visual,
             )
         )
 
     nextpos = 1
     for vector in vectors:
-        doc_id_split = vector.doc_id.id.split("/")
-        split = None
-        if len(doc_id_split) == 5:
-            rid, field_type, field, index, position = doc_id_split
-            paragraph_id = f"{rid}/{field_type}/{field}/{position}"
-        elif len(doc_id_split) == 6:
-            rid, field_type, field, split, index, position = doc_id_split
-            paragraph_id = f"{rid}/{field_type}/{field}/{split}/{position}"
-        else:
+        try:
+            vector_id = VectorId.from_string(vector.doc_id.id)
+        except (IndexError, ValueError):
             logger.warning(f"Skipping invalid doc_id: {vector.doc_id.id}")
             continue
-        start, end = position.split("-")
         merged_paragraphs.insert(
             nextpos,
-            TempFindParagraph(
-                vector_index=vector,
-                rid=rid,
-                field=f"/{field_type}/{field}",
+            TextBlockMatch(
+                paragraph_id=ParagraphId.from_vector_id(vector_id),
                 score=vector.score,
-                start=int(start),
-                end=int(end),
-                split=split,
-                id=paragraph_id,
+                score_type=SCORE_TYPE.VECTOR,
+                order=0,  # NOTE: this will be filled later
+                text="",  # NOTE: this will be filled later too
+                position=TextPosition(
+                    page_number=vector.metadata.position.page_number,
+                    index=vector.metadata.position.index,
+                    start=vector_id.vector_start,
+                    end=vector_id.vector_end,
+                    start_seconds=[x for x in vector.metadata.position.start_seconds],
+                    end_seconds=[x for x in vector.metadata.position.end_seconds],
+                ),
+                # TODO: get labels from index
+                field_labels=[],
+                paragraph_labels=[],
```

Comment:
is it a problem if this is not populated here?

Reply:
It's not populated anywhere, but it wasn't either. There was a TODO saying "get labels from index"